src/ngraph_encapsulate_op.cc

```diff
@@ -661,60 +643,100 @@ class NGraphEncapsulateOp : public OpKernel {
-        continue;
-      }
+    std::vector<TensorShape> input_shapes;
+    std::vector<const Tensor*> static_input_map;
+    std::shared_ptr<ngraph::Function> ng_function;
+    std::shared_ptr<ngraph::runtime::Executable> ng_exec;
+    ng::runtime::Backend* op_backend;
 
-      string ref_var_name = NGraphCatalog::GetInputVariableSharedName(
-          m_graph_id, def().name(), input_index);
-      NGraphVar* var;
-      OP_REQUIRES_OK(ctx, ctx->resource_manager()->Lookup<NGraphVar>(
-                              ctx->resource_manager()->default_container(),
-                              ref_var_name, &var));
+    // Get ngraph executable and inputs information
+    OP_REQUIRES_OK(ctx, GetNgExec(ctx, ng_exec, input_shapes, static_input_map,
+                                  op_backend));
 
-      if (var->need_sync_ng_tensor()) {
-        number_of_copies++;
-        copy_log_str << "Var_Sync[" << input_index << "] ";
-        ngraph::Event event_sync_ng_tf_tensors(
-            "Output: ng_tensor and tf_tensor sync", name(), "");
+    NGRAPH_VLOG(4)
+        << "NGraphEncapsulateOp::Compute got ngraph executable for cluster "
+        << m_ngraph_cluster;
 
-        NGRAPH_VLOG(4) << "In NGEncapsulate, ng tensor behind, needs to sync "
-                          "with tf-tensor";
-        WriteNGTensor(var->ng_tensor(), var->tensor());
-        // TODO(malikshr): We will be able to set the sync_ng_tensor to false
-        // once we do topological sort to add attributes like copy_to_tf
-        event_sync_ng_tf_tensors.Stop();
-        ngraph::Event::write_trace(event_sync_ng_tf_tensors);
-      }
+    int time_func_create_or_lookup = function_lookup_or_create.ElapsedInMS();
+    event_func_maybe_create.Stop();
 
-      ng_inputs[input_index] = var->ng_tensor();
+    NGRAPH_VLOG(4) << "NGraphEncapsulateOp::Compute got graph for cluster "
+                   << m_ngraph_cluster;
 
-      var->Unref();
+    Timer create_or_lookup_tensors;
+
+    if (m_freshness_tracker == nullptr) {
+      auto creator = [](NGraphFreshnessTracker** tracker) {
+        *tracker = new NGraphFreshnessTracker();
+        return Status::OK();
+      };
+      OP_REQUIRES_OK(
+          ctx, ctx->resource_manager()->LookupOrCreate<NGraphFreshnessTracker>(
+                   ctx->resource_manager()->default_container(),
+                   "ngraph_freshness_tracker", &m_freshness_tracker, creator));
     }
-    event_input_check_in_catalog.Stop();
-    ngraph::Event::write_trace(event_input_check_in_catalog);
+
+    NGRAPH_VLOG(4)
+        << "NGraphEncapsulateOp::Compute got freshness tracker for cluster "
+        << m_ngraph_cluster;
+
+#if defined(NGRAPH_TF_ENABLE_VARIABLES_AND_OPTIMIZERS)
+    bool log_copies = false;
+    OP_REQUIRES_OK(ctx, IsCopyLogEnabled(m_graph_id, log_copies));
+    copy_log_str << "KERNEL[" << type_string() << "]: " << name()
+                 << " ,GraphID " << m_graph_id << "\n";
 #endif
```

Comment:
Once this PR goes in , https://github.com/tensorflow/ngraph-bridge/pull/33, we can remove the #if defined around this log. We will have less if-defs in compute